qa/workunits/rbd/rbd_mirror_group_simple.sh

```diff
@@ -2939,6 +2939,72 @@ test_force_promote_before_initial_sync()
   start_mirrors "${secondary_cluster}"
 }
 
+# test snapshot sync to secondary cluster while restarting secondary cluster
+declare -a test_snap_sync_while_restarting_secondary_1=("${CLUSTER2}" "${CLUSTER1}" "${pool0}" 3 2G)
+
+test_snap_sync_while_restarting_secondary_scenarios=1
+
+test_snap_sync_while_restarting_secondary()
+{
+  local primary_cluster=$1 ; shift
+  local secondary_cluster=$1 ; shift
+  local pool=$1 ; shift
+  local image_count=$1 ; shift
+  local image_size=$1 ; shift
+  local image_size_bytes=$(numfmt --from=iec "$image_size")
+  local write_count=$((image_size_bytes / (4096*4096)))
+  local max_images
+  local group_snap_id
+
+  local group0=test-group0
+  local image_prefix="test_image"
+
+  echo "starting daemon on primary"
+  start_mirrors "${primary_cluster}"
+
+  echo "starting daemon on secondary"
+  start_mirrors "${secondary_cluster}"
+
+  group_create "${primary_cluster}" "${pool}/${group0}"
+  images_create "${primary_cluster}" "${pool}/${image_prefix}" "${image_count}" "${image_size}"
+  group_images_add "${primary_cluster}" "${pool}/${group0}" "${pool}/${image_prefix}" "${image_count}"
+
+  mirror_group_enable "${primary_cluster}" "${pool}/${group0}"
+
+  wait_for_group_present "${secondary_cluster}" "${pool}" "${group0}" "${image_count}"
+  wait_for_group_replay_started "${secondary_cluster}" "${pool}"/"${group0}" "${image_count}"
+  wait_for_group_status_in_pool_dir "${secondary_cluster}" "${pool}"/"${group0}" 'up+replaying' "${image_count}"
+
+  wait_for_group_synced "${primary_cluster}" "${pool}"/"${group0}" "${secondary_cluster}" "${pool}"/"${group0}"
+
+  max_images=$((image_count-1))
```

Comment:
I don't think we just need an additional variable `max_images` here

`for i in $(seq 0 $(("${image_count}"-1))); do`